### PR TITLE
Reword and clean up issue templates

### PR DESCRIPTION
**Pull request by dhruvkb**
## Description
This PR 
- cleans up the wording of the issue templates
- updates the format to match the recently updated PR template
- allows reporters to indicate their inclination to work on that issue

## Technical details
The checkbox can also be used to setup automation that assigns the reporter to the issue as soon as the awaiting 'triage label' is replaced with anything other than 'invalid' or 'wontfix'.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -4,32 +4,33 @@ about: Create a report to help us improve
...
- - Browser (ex. chrome, safari)
- - Version (ex. 22)
-
-## Smartphone (please complete the following information)
- - Device: (ex. iPhone6)
- - OS: (ex. iOS8.1)
- - Browser (ex. stock browser, safari)
- - Version (ex. 22)
+## Environment
+<!-- If the problem is environment specific, please complete this. -->
```
Comment (mathemancer) [inline .github/ISSUE_TEMPLATE/bug_report.md:24]: I'd rather the assumption be that environment is relevant:
```suggestion
<!-- Please complete this, unless you are certain the problem is not environment specific -->
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 2]
```diff
@@ -4,14 +4,18 @@ about: Suggest an idea for this project
 labels: enhancement, awaiting triage
 ---
 
-## Problem Description
-<!-- Please provide a clear and concise description of what the problem or issue the proposed feature will solve (Ex. I'm always frustrated when ...) -->
+## Problem
+<!-- Describe a problem solved by this feature; or delete the section entirely. -->
```
Comment (mathemancer) [inline .github/ISSUE_TEMPLATE/feature_request.md:8]: I'm not sure this should be optional

Comment (kgodey) [reply .github/ISSUE_TEMPLATE/feature_request.md:8]: I don't think all features necessarily solve problems, they are sometimes new capabilities.

**Comment by dhruvkb** [comment]
@kgodey made the change.

**Comment by kgodey** [comment]
thanks @dhruvkb!

**Pull request merged**